File: src/employeemanager/model/PartTimeEmployee.java
package employeemanager.model;

import employeemanager.exceptions.InvalidDataException;
import java.time.LocalDate;

public class PartTimeEmployee extends Employee {

    private double hourlyRate;
    private int maxHoursPerWeek;
    private LocalDate contractEndDate;

    public PartTimeEmployee(int employeeId,
                            String firstName,
                            String lastName,
                            LocalDate startDate,
                            LocalDate dateOfBirth,
                            double hourlyRate,
                            int maxHoursPerWeek,
                            LocalDate contractEndDate) throws InvalidDataException {
        super(employeeId, firstName, lastName, startDate, dateOfBirth);

        if (hourlyRate < 0) {
            throw new InvalidDataException("Hourly rate cannot be negative.");
        }

        this.hourlyRate = hourlyRate;
        this.maxHoursPerWeek = maxHoursPerWeek;
        this.contractEndDate = contractEndDate;
    }

    @Override
    public double getPay() {
        return hourlyRate * maxHoursPerWeek;
    }

    @Override
    public void reportToManager() {
        System.out.println("[PartTime] " + getFullName() + " reporting to manager.");
        super.reportToManager();
    }
}

File: Part 3/EmployeeManager/models/PartTimeEmployee.cs
using System;
using InvalidDataException = EmployeeManager.Exceptions.InvalidDataException;

namespace EmployeeManager.Models
{
    public class PartTimeEmployee : Employee
    {
        public double HourlyRate { get; set; }
        public int MaxHoursPerWeek { get; set; }
        public DateTime ContractEndDate { get; set; }

        public PartTimeEmployee(
            int employeeId,
            string firstName,
            string lastName,
            DateTime startDate,
            DateTime dateOfBirth,
            double hourlyRate,
            int maxHoursPerWeek,
            DateTime contractEndDate)
            : base(employeeId, firstName, lastName, startDate, dateOfBirth)
        {
            if (hourlyRate < 0)
                throw new InvalidDataException("Hourly rate cant be negative");

            HourlyRate = hourlyRate;
            MaxHoursPerWeek = maxHoursPerWeek;
            ContractEndDate = contractEndDate;
        }

        public override double GetPay() => HourlyRate * MaxHoursPerWeek;

        public override void ReportToManager()
        {
            Console.WriteLine($"[PartTime] {FullName} reporting to manager");
            base.ReportToManager();
        }
    }
}
